### DISCUSSION TITLE
Mockup: Migrate Module Settings into Persona Bar

### DISCUSSION THREAD

**Discussion by Chris-Behling** [dnnsoftware/Dnn.Platform#4271] [Show and tell]
I’m submitting a few mock-ups I’ve created to hopefully spark interest in migrating the Module Settings controls into the Persona Bar to provide a consistent management experience. In my opinion DNN should phase out the old pop-up dialog for managing modules and content.

The current pop-up dialog is from an older version of DNN and has a completely different UI from the Persona Bar and creates inconsistency in the platform.

[Module Settings -- New](https://www.dropbox.com/sh/81juz9uumk3dgo9/AACpOfm5TDR71WDLQoB-9EeJa?dl=0&preview=Module-Setting--new.jpg)

[Permissions Settings -- New](https://www.dropbox.com/sh/81juz9uumk3dgo9/AACpOfm5TDR71WDLQoB-9EeJa?dl=0&preview=Permissions--new.jpg)

[Page Settings -- New](https://www.dropbox.com/sh/81juz9uumk3dgo9/AACpOfm5TDR71WDLQoB-9EeJa?dl=0&preview=Page-Settings--new.jpg)

I'd be happy to discuss this more. One area that comes to mind is when 3rd party modules, and some core modules, that have their own tab for module settings. I'm not entirely sure how DNN injects this content and to go even further, how much customization users can place in here (color picker for example).  

![Module-Setting--new](https://user-images.githubusercontent.com/47400126/98007515-172e5600-1db9-11eb-9d7a-faf5b60d85dc.jpg)

![Permissions--new](https://user-images.githubusercontent.com/47400126/98007529-1d243700-1db9-11eb-8f89-8ee59520ea5c.jpg)

![Pages-Settings--new](https://user-images.githubusercontent.com/47400126/98007556-257c7200-1db9-11eb-9e84-a4824f0773d7.jpg)

**Comment by valadas** [comment]
This would be awesome, for the 4th tab, currently 3rd parties provide a webforms control for the module specific settings, we could maybe load that up into the 4th tab somehow, but it would be great to at some point deprecated that webforms only control and provide a new manifest node that would have universal module settings as html/css/js principle just like in the QuickSettings but that would load up in the Persona Bar too. That however opens another can of worms because the PersonaBar is an extension and can be replaced by 3rd parties by their own custom control panel, so we also need to keep that in mind. I think i would prefer to load this up in a modal than in the persona bar as I think of it...

**Reply by Chris-Behling** [reply]
If the 4th tab was shifted over to support/build with a manifest file could the same (current) user experience with the pop-up be kept for those who are not using the PersonaBar? Then for those that are using the PersonaBar it instead builds the view inside the PersonaBar?

Are there any estimates on what percentage of DNN installs on 9.0.0+ are not using the PersonaBar?

I'd have to imagine that if someone is going as far to remove the PersonaBar they are likely also modifying the UI of the Module Settings.

**Comment by valadas** [comment]
We don't really have a way to know, but at any rate, we need some sort of transition plan to not break all the module settings. The actual implementation will depend on how the PR comes in and this is a major task to write in. I know very very few modules have implemented QuickSettings as one example, so if we change how that works, we need to at least let 3rd parties know about the new way and give them time to reimplement. Some modules have huge module settings that rely on webforms.

Don't get me wrong, I am totally in favor of a module settings revamp just discussion how to transition to it in a way that does not brake all modules and people who might use a 3rd party control panel.
